lib/adagrams.rb

```diff
@@ -0,0 +1,134 @@
+# Sources: https://stackoverflow.com/questions/33160746/is-there-anything-bsearch-for-strings-in-ruby
+
+require 'csv'
+
+ALL_LETTERS = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1
+}
+
+SCORE_CHART = {
+    A: 1,
+    B: 3,
+    C: 3,
+    D: 2,
+    E: 1,
+    F: 4,
+    G: 2,
+    H: 4,
+    I: 1,
+    J: 8,
+    K: 5,
+    L: 1,
+    M: 3,
+    N: 1,
+    O: 1,
+    P: 3,
+    Q: 10,
+    R: 1,
+    S: 1,
+    T: 1,
+    U: 1,
+    V: 4,
+    W: 4,
+    X: 8,
+    Y: 4,
+    Z: 10
+}
+
+def draw_letters
+  all_letters = ALL_LETTERS.map do |keys, values|
+    [keys.to_s] * values    #follow-up, why do square brackets work??
+  end
+
+  return all_letters.flatten.shuffle.first(10)
+end
+
+def count_letters(letters_array)
+  letter_count = {}
+
+  letters_array.each do |letter|
+    (letter_count.include? letter) ? letter_count[letter] += 1 : letter_count[letter] = 1
+  end
+
+  return letter_count
+end
+
+def uses_available_letters?(input, letters_in_hand)
+  hand_count = count_letters(letters_in_hand)
+  input_count = count_letters(input.split(''))
```

Comment:
`chars` is a slightly clearer way to do this:

```suggestion
  input_count = count_letters(input.chars)
```